Objects/typeobject.c

```diff
@@ -285,9 +285,9 @@ PyType_ClearCache(void)
 void
 _PyType_Fini(PyInterpreterState *interp)
 {
-    _PyType_ClearCache(&interp->type_cache);
```

Comment:
This change leaks memory: type_cache_clear() clears objects and should be done in subinterpreters.

I suggest to pass interp to _PyType_ClearCache() and check `_Py_IsMainInterpreter(interp)` in this function to decide if next_version_tag must be modified or not.

Reply:
Thank you for your suggestion, it is very reasonable. I have made the changes as you suggested.